prometheus_juju_exporter/collector.py

```diff
@@ -185,14 +213,15 @@ def _get_container_status(
         for container in containers.values():
             value = self._get_gauge_value(container["agent-status"]["status"])
 
+            hostname = self._get_machine_sensible_hostname(container)
             labels = self._create_gauge_label(
-                hostname=container["hostname"],
+                hostname=hostname,
                 model_name=model_name,
                 machine_type=MachineType.LXD.value,
             )
 
-            if labels["hostname"] and labels["hostname"] != "None":
-                self.currently_cached_labels[container["hostname"]] = (
+            if labels["hostname"] != "None":
```

Comment:
Same as the previous comment. It feels weird to use labels["hostname"} in some places and `hostname` in others if they are the same thing.

Reply:
addressed this in the update